cmd/kubeadm/app/util/cgroupdriver.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	utilsexec "k8s.io/utils/exec"
+)
+
+func validateCgroupDriver(driver string) error {
+	if driver != "cgroupfs" && driver != "systemd" {
+		return fmt.Errorf("unknown cgroup driver %q", driver)
+	}
+	return nil
+}
+
+func callDockerInfo() (string, error) {
+	execer := utilsexec.New()
+	out, err := execer.Command("docker", "info").Output()
```

Comment:
Add TODO to be able to run only `docker info -f "{{.CgroupDriver}}"` when we can require docker 1.13+

Reply:
Can't we get this from the critool at all?  What happens when folks pick another CRI?

Reply:
> Can't we get this from the critool at all?

no, to my understanding and according to the help filbranden gave us here:
https://github.com/kubernetes/kubeadm/issues/844#issuecomment-391159929

this definitely needs extensions down the line for other CRI, but not doable for 1.11.
the main faults here come from crictl not supporting this and a certain kubelet entanglement.

> What happens when folks pick another CRI?

it would pick the docker driver and if that driver doesn't match the other CRI driver and the kubelet driver the user will get an error. in that case he/she will have to manually override the kubelet driver.

not ideal, but this PR is the offspring of the issue discussion.